units_cpp/player.cc

```diff
@@ -1,65 +1,75 @@
 #include "player.h"
+
 #include <algorithm>
-#include "unit_kind.h"
+
 #include "game_rules.h"
 #include "json.h"
+#include "unit_kind.h"
+
 
-Player::Player() : alive(true), coins(0), next_unit(0) {}
+Player::Player():
+		alive(true),
+		coins(0),
+		next_unit(0)
+{}
 
 Player::Player(const GameRules &rules) : alive(true), coins(0), next_unit(0) {
-  std::vector<UnitKind*> starting_units = rules.get_starting_units();
-  for (auto it = starting_units.begin(); it != starting_units.end(); it++) {
-    build_unit(**it);
-  }
+	std::vector<UnitKind*> starting_units = rules.get_starting_units();
+	for (auto it = starting_units.begin(); it != starting_units.end(); it++) {
+		build_unit(**it);
+	}
 }
 
-Player::Player(const Player &player)
-    : units(player.units)
-    , alive(player.alive)
-    , coins(player.coins)
-    , tech(player.tech)
-    , next_unit(player.next_unit) {}
+Player::Player(const Player &player):
+		units(player.units),
+		alive(player.alive),
+		coins(player.coins),
+		tech(player.tech),
+		next_unit(player.next_unit)
```

Comment:
Comma-first style is an established thing that Charles and I have used and grown somewhat fond of. It may seem ugly at first glance from your experience with the comma in English writing, but it has distinct advantages in programming. See this gist for some of the motivation behind it: https://gist.github.com/isaacs/357981

I'll be changing this back.